### PR TITLE
Avoid including vectorization.h in a too central place

### PR DESCRIPTION
**Pull request by kronbichler**
Due to the change in #4311 we might now potentially issue errors in user projects that do not use `VectorizedArray` at all. We can avoid them by not including `VectorizedArray` in any header file except for the `matrix_free/` subdirectory where we must indeed make sure that things work fine. With this change the places left are only `templates.h` files, so the error message from #4311 does exactly appear where it should. To fix the test `elasticity-kinematics_02`, I had to add a few forward declarations.

### CODE REVIEW

File: include/deal.II/base/symmetric_tensor.h [hunk 1]
```diff
@@ -21,7 +21,6 @@
 #include <deal.II/base/numbers.h>
 #include <deal.II/base/table_indices.h>
 #include <deal.II/base/template_constraints.h>
-#include <deal.II/base/vectorization.h>
```
Comment (davydden) [inline include/deal.II/base/symmetric_tensor.h]: hm, this was added in https://github.com/dealii/dealii/pull/3967 do the test still work for you?

**Comment by davydden** [comment]
> We can avoid them by not including VectorizedArray in any header file except for the matrix_free/

i think you are breaking other tests by this

**Comment by kronbichler** [comment]
@davydden I ran the test suite and everything seems fine. This was fixed by commit 46c1c80 or PR #4059 as far as I can tell.

**Comment by davydden** [comment]
on the second look, i think you are right. `symmetric_tensor.h` does not have to know about `VectorizedArray`, it's only when you want to use both together, as done in `tests/base/symmetric_tensor_39.cc`, you would include both headers and instantiate `SymmetricTensor` for `VectorizedArray`.

**Pull request merged**